site/src/Librecores/ProjectRepoBundle/Doctrine/ProjectMetricsProvider.php

```diff
@@ -217,25 +219,45 @@ function (LanguageStat $a, LanguageStat $b) {
             }
         );
 
-        /** @var LanguageStat[] $topLangs */
-        $topLangs = array_slice($langStats, 0, 4);
-
-        $result = [];
-        if (count($langStats) > 4) {
-            $others = 0;
+        $minValue = 0.05 * $langStats[0]->getFileCount();
 
-            for ($i = 4; $i < count($langStats); $i++) {
-                $others += $langStats[$i]->getLinesOfCode();
+        $result = ['Others' => 0];
+        foreach ($langStats as $lang) {
+            $fc = $lang->getFileCount();
+            if ($fc < $minValue) {
+                $result['Others'] += $fc;
+            } else {
+                $result[$lang->getLanguage()] = $fc;
             }
-            $result['others'] = $others;
         }
 
-        foreach ($topLangs as $lang) {
-            $result[$lang->getLanguage()] = $lang->getLinesOfCode();
-        }
-
-        sort($result, SORT_NUMERIC);
-
         return $result;
     }
+
+    /**
+     * Get a histogram of contributors over a range of dates
```

Comment:
What exactly is a "histogram of contributors"? A contributor (in my understanding) is a person, and I'm not sure how to interpret a histogram of people... Could you rephrase please?